components/cam/src/physics/cam/shoc_intr.F90

```diff
@@ -910,46 +910,46 @@ subroutine shoc_tend_e3sm( &
 
    enddo  ! end time loop
    
-   ! Arrays need to be "flipped" to E3SM grid
+   ! Transfer back to pbuf variables
    
    do k=1,pver
      do i=1,ncol 
-       um(i,k) = um_in(i,pver-k+1)
-       vm(i,k) = vm_in(i,pver-k+1)
-       thlm(i,k) = thlm_in(i,pver-k+1)
-       rtm(i,k) = rtm_in(i,pver-k+1)
-       rcm(i,k) = rcm_shoc(i,pver-k+1)
-       cloud_frac(i,k) = min(cloudfrac_shoc(i,pver-k+1),1._r8)
-       wthv(i,k) = wthv_in(i,pver-k+1)
-       tke(i,k) = tke_in(i,pver-k+1)
+       um(i,k) = um_in(i,k)
+       vm(i,k) = vm_in(i,k)
+       thlm(i,k) = thlm_in(i,k)
+       rtm(i,k) = rtm_in(i,k)
+       rcm(i,k) = rcm_shoc(i,k)
+       cloud_frac(i,k) = min(cloudfrac_shoc(i,k),1._r8)
+       wthv(i,k) = wthv_in(i,k)
+       tke(i,k) = tke_in(i,k)
        
        do ixind=1,edsclr_dim
-         edsclr_out(i,k,ixind) = edsclr_in(i,pver-k+1,ixind)
+         edsclr_out(i,k,ixind) = edsclr_in(i,k,ixind)
        enddo      
 
-       shoc_mix(i,k) = shoc_mix_out(i,pver-k+1)
-       tk(i,k) = tk_in(i,pver-k+1)
-       tkh(i,k) = tkh_in(i,pver-k+1)
-       isotropy(i,k) = isotropy_out(i,pver-k+1)
-       w_sec(i,k) = w_sec_out(i,pver-k+1)
-       wqls(i,k) = wqls_out(i,pver-k+1)
-       brunt(i,k) = brunt_out(i,pver-k+1)
+       shoc_mix(i,k) = shoc_mix_out(i,k)
+       tk(i,k) = tk_in(i,k)
+       tkh(i,k) = tkh_in(i,k)
+       isotropy(i,k) = isotropy_out(i,k)
+       w_sec(i,k) = w_sec_out(i,k)
+       wqls(i,k) = wqls_out(i,k)
+       brunt(i,k) = brunt_out(i,k)
  
      enddo
    enddo
 
    do k=1,pverp
      do i=1,ncol
 
-       w3(i,k) = w3_out(i,pverp-k+1)
-       thl_sec(i,k) = thl_sec_out(i,pverp-k+1)
-       qw_sec(i,k) = qw_sec_out(i,pverp-k+1)
-       qwthl_sec(i,k) = qwthl_sec_out(i,pverp-k+1)
-       wthl_sec(i,k) = wthl_sec_out(i,pverp-k+1)
-       wqw_sec(i,k) = wqw_sec_out(i,pverp-k+1)
-       wtke_sec(i,k) = wtke_sec_out(i,pverp-k+1)
-       uw_sec(i,k) = uw_sec_out(i,pverp-k+1)
-       vw_sec(i,k) = vw_sec_out(i,pverp-k+1)
+       w3(i,k) = w3_out(i,k)
```

Comment:
Same for the output variables. Now that shoc agrees with e3sm, I think we should be able to avoid all of these copies.